hbase-hadoop-compat/src/main/java/org/apache/hadoop/hbase/replication/regionserver/MetricsReplicationSourceSource.java

```diff
@@ -76,4 +77,13 @@
   long getWALEditsRead();
   long getShippedOps();
   long getEditsFiltered();
+  /**
+   * Sets the total usage of memory used by edits in memory read from WALs.
+   * @param usage The memory used by edits in bytes
+   */
+  void setWALReaderEditsBufferBytes(long usage);
```

Comment:
if we only want the global metric, why add this here instead of just on `MetricsReplicationGlobalSourceSource`?

Reply:
IIRC, the type hierarchy is terrible and prevented this from being done in a more clean way (that wouldn't require an unsafe cast), but let me double-check that.

Reply:
Yeah, so the reason this was done this was is that we had no interface for global metrics, just a second implementation of the per-source interface.

Stubbed in a new interface so that we can have "global-metrics-only" methods that don't pollute per-source metrics.